e2eTests/e2eTestHelper.psm1

```diff
@@ -346,13 +346,27 @@ function CreateAlGoRepository {
     if (!$template.Contains('@')) {
         $template += '@main'
     }
+    $waitMinutes = Get-Random -Minimum 0 -Maximum 4
+    $templateFolder = ''
+    if ($template.Contains('|')) {
+        # In order to run tests on the direct AL-Go Development branch, specify the folder in which the template is located after a | character in template
+        # example: "https://github.com/freddydk/AL-Go@branch|Templates/Per Tenant Extension"
+        $templateFolder = $template.Split('|')[1]

```

Comment:
What's the reasoning for having the template folder as part for the template parameter?
Isn't true that the template folder can be either `Templates/Per Tenant Extension" or "Templates/AppSource App"?

If yes, then another parameter $templateType (with possible values "PTE" and "AppSource App") sounds like a better approach.

Also, whether you keep that design or not, consider using a regex to match the different types:
E.g. `$template -match "https://github.com/(.*)/AL-Go@(.*)\|(.*)`.

Reply:
If we are running E2E tests locally, we need to set $global:pteTemplate and $global:appSourceTemplate (which are default values to the parameters in the E2E tests (https://github.com/microsoft/AL-Go/blob/6bc6da7de33a23ed410714f7db7c7e36d62f561d/e2eTests/scenarios/UseProjectDependencies/runtest.ps1#L6C5-L6C5))
For normal E2E test running, there will not be a TemplateFolder and I didn't want to add a parameter more all over the place, just in order to be able to support running locally on direct AL-Go repo.

Reply:
Have added this info to Contribute.md + an explanationation of what the various variables needs to be set to

Reply:
I think I got it now.

But in that case, the template folder can be determined directly, right?
If you set `$global:pteTemplate` to `freddydk/AL-Go@main\|Templates/Per Tenant Extension` then the template folder after `|` is redundant. Because it's easy to determine that it's direct AL-Go and it's a PTE template (_UpdateALGoSystemFiles_ does it).

Also, in the file you mentioned there's this line
[$template = "https://github.com/$($pteTemplate)@main"](https://github.com/microsoft/AL-Go/blob/6bc6da7de33a23ed410714f7db7c7e36d62f561d/e2eTests/scenarios/UseProjectDependencies/runtest.ps1#L52).
Reading the doc, wouldn't `$template` then be `https://github.com/freddydk/AL-Go@main\|Templates/Per Tenant Extension@main`?

Reply:
Yeah, well this is the code that creates the repo - it doesn't know which kind of repo it is until it has downloaded the folder which the templateFolder points out.

Reply:
fixed with the latest checkin - @main gets added later

Reply:
Ok then, resolving.